javaparser-symbol-solver-core/src/main/java/com/github/javaparser/symbolsolver/logic/AbstractTypeDeclaration.java

```diff
@@ -43,23 +47,43 @@
         Set<String> methodsSignatures = new HashSet<>();
 
         for (ResolvedMethodDeclaration methodDeclaration : getDeclaredMethods()) {
-            methods.add(new MethodUsage(methodDeclaration));
-            methodsSignatures.add(methodDeclaration.getSignature());
+            MethodUsage methodUsage = new MethodUsage(methodDeclaration);
+            methods.add(methodUsage);
+            methodsSignatures.add(getSignature(methodUsage));
         }
 
         for (ResolvedReferenceType ancestor : getAllAncestors()) {
-            for (MethodUsage mu : ancestor.getDeclaredMethods()) {
-                String signature = mu.getDeclaration().getSignature();
+            List<Pair<ResolvedTypeParameterDeclaration, ResolvedType>> typeParametersMap = ancestor.getTypeParametersMap();
+            for (MethodUsage ancestorMethodUsage : ancestor.getDeclaredMethods()) {
+                MethodUsage methodUsage = ancestorMethodUsage;
+                for (Pair<ResolvedTypeParameterDeclaration, ResolvedType> p : typeParametersMap) {
+                    methodUsage = methodUsage.replaceTypeParameter(p.a, p.b);
+                }
+                String signature = getSignature(methodUsage);
                 if (!methodsSignatures.contains(signature)) {
                     methodsSignatures.add(signature);
-                    methods.add(mu);
+                    methods.add(ancestorMethodUsage);
                 }
             }
         }
 
         return methods;
     }
 
+    private String getSignature(MethodUsage mu) {
```

Comment:
Is there a reason for using a custom private method here rather than using the `getMethodSignature()` method?

My presumption is the answer is "yes" and that there's something I'm currently not seeing :)

Reply:
I coulnd't find a `getMethodSignature()` method, did you mean why I didn't use the `methodDeclaration.getSignature()` method?

To find overridden generic methods, I need a signature where the generic types have been replaced with the information from the subclass.

For instance, consider the example:

```
public static interface Foo<S, T> extends Function<S, T> {

    T foo(S str);

    @Override
    default T apply(S str) {
        return foo(str);
    }
}
```

If we use the `methodDeclaration.getSignature()` method, we have the following signatures:
apply(S), foo(S) (from Foo) and apply(T) (from Function)

But we don't want the apply(T) here since it's overridden by apply(S).
For that, I've copied the logic from `methodDeclaration.getSignature()` into a private method and execute it on a `MethodUsage `object (and not on a `MethodDeclaration`).

That method could probably be added as a public method to MethodUsage, but I'm not sure it is usable at other places...

Reply:
> I coulnd't find a `getMethodSignature()` method, did you mean why I didn't use the `methodDeclaration.getSignature()` method?

Apologies, yes you are correct! Typo on my part!